addons/point_of_sale/models/pos_config.py

```diff
@@ -633,7 +633,7 @@ def open_ui(self):
         """
         self.ensure_one()
         # In case of test environment, don't create the pdf
-        if self.env.su and not tools.config['test_enable']:
+        if self.env.uid == SUPERUSER_ID and not tools.config['test_enable']:
```

Comment:
what is the difference? 
https://github.com/odoo/odoo/blob/79ebb9abcaa4c86b2ff7fc251b80c0f8c11dfdc7/odoo/api.py#L553-L554

Reply:
@vlst-odoo 

The difference is that when it's called from the controller, the user is **SUPERUSER**. We can resolve this by invoking `open_ui()` as `pos_config.with_user(request.env.user).open_ui()`. Indeed this fix needs to be applied everywhere `open_ui()` is called, including in custom modules. So it is better to fix it here!

What do you think?